lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,79 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user_header)
+          Thread.current[:attr_current_user] = new(user_header)
+        end
+
+        def self.with_logged_in_user(user_header)
+          alternate_user = new(user_header)
+          yield alternate_user
+        ensure
+          alternate_user = nil
+        end
+
+        def initialize(user_header)
+          @user = user_header
+        end
+
+        def username
```

Comment:
@syncrou How would this work if the 3 scale team or the RHN login adds a new field. Would we have to update this code or would we have direct access to the raw hash to handle on-going enhancements from outside of us.

In the IPP there is a nested hash is they key names get duplicated in the nested hash what would these functions return?

Reply:
@mkanoor I advocate we explicitly expose selected attributes rather than blindly pass through the raw hash. The consumer of this class should rely on the class contract rather than the backend (3scale or RHN) document. 

@syncrou shouldn't `tenant` be exposed too? I saw your existing work directly examined the header to fetch tenant. Is it better to get tenant through the user object?

Reply:
@bzwei - Yeah, I'll get that added.